Mathlib/CategoryTheory/Triangulated/Opposite/Functor.lean

```diff
@@ -0,0 +1,216 @@
+with the equivalences `Pretriangulated.triangleOpEquivalence` on `C` and `D`.
+-/
+noncomputable def mapTriangleOpCompTriangleOpEquivalenceFunctor :
+    F.mapTriangle.op ⋙ (triangleOpEquivalence D).functor ≅
+      (triangleOpEquivalence C).functor ⋙ F.op.mapTriangle :=
+  NatIso.ofComponents
+    (fun T ↦ F.mapTriangleOpCompTriangleOpEquivalenceFunctorApp T.unop)
+    (by intros; ext <;> dsimp <;> simp only [comp_id, id_comp])
+
+/--
+If `F : C ⥤ D` commutes with shifts, this is the 2-commutative square of categories
+`CategoryTheory.Functor.mapTriangleOpCompTriangleOpEquivalenceFunctor`.
+-/
+noncomputable instance :
+    CatCommSq (F.mapTriangle.op) (triangleOpEquivalence C).functor
+      (triangleOpEquivalence D).functor F.op.mapTriangle :=
+  ⟨F.mapTriangleOpCompTriangleOpEquivalenceFunctor⟩
+
+/--
+Vertical inverse of the 2-commutative square of
+`CategoryTheory.Functor.mapTriangleOpCompTriangleOpEquivalenceFunctor`.
+-/
+noncomputable instance :
+    CatCommSq (F.op.mapTriangle) (triangleOpEquivalence C).inverse
+      (triangleOpEquivalence D).inverse F.mapTriangle.op :=
+  CatCommSq.vInv (F.mapTriangle.op) (triangleOpEquivalence C)
+      (triangleOpEquivalence D) F.op.mapTriangle inferInstance
+
+/--
+If `F : C ⥤ D` commutes with shifts, this expresses the compatibility of `F.mapTriangle`
+with the equivalences `Pretriangulated.triangleOpEquivalence` on `C` and `D`.
+-/
+noncomputable def opMapTriangleCompTriangleOpEquivalenceInverse :
+    F.op.mapTriangle ⋙ (triangleOpEquivalence D).inverse ≅
+      (triangleOpEquivalence C).inverse ⋙ F.mapTriangle.op :=
+  CatCommSq.iso (F.op.mapTriangle) (triangleOpEquivalence C).inverse
+      (triangleOpEquivalence D).inverse F.mapTriangle.op
+
+/-- If `F` is triangulated, so is `F.op`.
+-/
+lemma isTriangulated_op [F.IsTriangulated] : F.op.IsTriangulated where
+  map_distinguished T dT := by
+    rw [mem_distTriang_op_iff]
+    exact Pretriangulated.isomorphic_distinguished _
+      ((F.map_distinguished _ (unop_distinguished _ dT))) _
+      (((opMapTriangleCompTriangleOpEquivalenceInverse F).symm.app T).unop)
+
+/-- If `F.op` is triangulated, so is `F`.
+-/
+lemma isTriangulated_of_op [F.op.IsTriangulated] : F.IsTriangulated where
+  map_distinguished T dT := by
+    have := distinguished_iff_of_iso ((triangleOpEquivalence D).unitIso.app
+      (Opposite.op (F.mapTriangle.obj T))).unop
+    rw [Functor.id_obj, Opposite.unop_op (F.mapTriangle.obj T)] at this
+    rw [← this, Functor.comp_obj, ← mem_distTriang_op_iff, ← Functor.op_obj, ← Functor.comp_obj,
+      distinguished_iff_of_iso ((mapTriangleOpCompTriangleOpEquivalenceFunctor F).app
+      (Opposite.op T))]
+    apply F.op.map_distinguished
+    have := distinguished_iff_of_iso ((triangleOpEquivalence C).unitIso.app (Opposite.op T)).unop
+    rw [Functor.id_obj, Opposite.unop_op T] at this
+    rw [← this, Functor.comp_obj, ← mem_distTriang_op_iff] at dT
+    exact dT
+
```

Comment:
Could you add  a lemma `isTriangulated_iff_op : F.op.IsTriangulated ↔ F.IsTriangulated`?

Reply:
I could.